### PR TITLE
GDB-5367 Refresh autocomplete status in rdf search box

### PR DESCRIPTION
**Pull request by yordanalexandrov**
Added a `autocompleteStatus` service which sets the autocomplete enabled status to local storage.
The service also monitors local storage for change (supporting different tabs) and broadcasts the status on change.
Added watchers for all search resource inputs.

### CODE REVIEW

File: src/js/angular/core/directives/rdfresourcesearch/rdf-resource-search.directive.js [hunk 1]
```diff
@@ -13,13 +13,23 @@ function rdfResourceSearchDirective($rootScope, $timeout,
...
               $scope.getActiveRepository())
               .success(function () {
-                $scope.getAutocompletePromise = AutocompleteRestService.checkAutocompleteStatus();
+                checkAutocompleteStatus();
+              });
+        }
+      }
+
+      function checkAutocompleteStatus() {
+        if (!autocompleteStatus) {
```
Comment (svilenvelikov) [inline src/js/angular/core/directives/rdfresourcesearch/rdf-resource-search.directive.js:29]: If `autocompleteStatus=false` then you are going to check it against the GDB and eventually if it's enabled you rise the flag. So far so good. But what happens if `autocompleteStatus=true` and the user tries to disable the autocomplete? You won't notice it here, will you?

Comment (svilenvelikov) [reply src/js/angular/core/directives/rdfresourcesearch/rdf-resource-search.directive.js:29]: Maybe better approach is to find if it's possible to expose the autocomplete status globally from the `AutocompleteCtrl` somehow, e.g. through a statefull service or through the global scope.

Comment (yordanalexandrov) [reply src/js/angular/core/directives/rdfresourcesearch/rdf-resource-search.directive.js]: Refactored code into a service. Added local storage key and watcher. When status is changed it is broadcasted in rootScope.

File: src/js/angular/core/services/autocomplete-status.service.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  function refreshAutocompleteStatus() {
+    AutocompleteRestService.checkAutocompleteStatus()
+        .success(function (autocompleteStatusResponse) {
+          that.setAutocompleteStatus(autocompleteStatusResponse);
+        });
+  }
+
+  $rootScope.$on('repositoryIsSet', refreshAutocompleteStatus);
+}
```
Comment (svilenvelikov) [inline src/js/angular/core/services/autocomplete-status.service.js:26]: Great. I like it.

**Review by svilenvelikov — APPROVED** [review]

**Pull request merged**